collector/perflib.go

```diff
@@ -0,0 +1,131 @@
+package collector
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/leoluk/perflib_exporter/perflib"
+	"github.com/prometheus/common/log"
+)
+
+func getPerflibSnapshot() (map[string]*perflib.PerfObject, error) {
+	objects, err := perflib.QueryPerformanceData("Global")
+	if err != nil {
+		return nil, err
+	}
+
+	indexed := make(map[string]*perflib.PerfObject)
+	for _, obj := range objects {
+		indexed[obj.Name] = obj
+	}
+	return indexed, nil
+}
+
+const (
```

Comment:
Most of these constants are exported by perflib_exporter: https://github.com/leoluk/perflib_exporter/blob/master/collector/mapper.go

(as of https://github.com/leoluk/perflib_exporter/pull/14)

Reply:
Ah, I was only looking at the perflib package.